src/Spago/Config.purs

```diff
@@ -164,11 +165,22 @@ type ReadWorkspaceOptions =
   , migrateConfig :: Boolean
   }
 
+type PrelimWorkspace =
+  { backend :: Maybe Core.BackendConfig
+  , buildOpts :: Maybe
+                 { censorLibraryWarnings :: Maybe Core.CensorBuildWarnings
+                 , output :: Maybe String
+                 , statVerbosity :: Maybe Core.StatVerbosity
+                 }
+  , extraPackages :: Maybe (Map PackageName Core.ExtraPackage)
+  , packageSet :: Maybe Core.SetAddress
+  }
+
 -- | Reads all the configurations in the tree and builds up the Map of local
 -- | packages to be integrated in the package set
 readWorkspace :: ∀ a. ReadWorkspaceOptions -> Spago (Registry.RegistryEnv a) Workspace
 readWorkspace { maybeSelectedPackage, pureBuild, migrateConfig } = do
-  logInfo "Reading Spago workspace configuration..."
+  logInfo "Reading spago.yaml..."
```

Comment:
i.m.o. this is more correct because it is reading a `spago.yaml` that may not contain a workspace.

Reply:
Sure